sdks/java/harness/src/main/java/org/apache/beam/fn/harness/debug/DataSampler.java

```diff
@@ -35,22 +34,22 @@
  * simultaneously, even if computing the same logical PCollection.
  */
 public class DataSampler {
+  private static final Logger LOG = LoggerFactory.getLogger(DataSampler.class);
 
   /**
    * Creates a DataSampler to sample every 1000 elements while keeping a maximum of 10 in memory.
    */
   public DataSampler() {
-    this.maxSamples = 10;
-    this.sampleEveryN = 1000;
+    this(10, 1000);
   }
 
   /**
    * @param maxSamples Sets the maximum number of samples held in memory at once.
    * @param sampleEveryN Sets how often to sample.
    */
   public DataSampler(int maxSamples, int sampleEveryN) {
-    this.maxSamples = maxSamples;
-    this.sampleEveryN = sampleEveryN;
+    this.maxSamples = maxSamples <= 0 ? 10 : maxSamples;
+    this.sampleEveryN = sampleEveryN <= 0 ? 1000 : sampleEveryN;
```

Comment:
Its usually better to throw an IllegalArgumentException in these cases instead of silently having a different behavior then before.

```suggestion
    checkArgument(maxSamples > 0, "Expected positive number of samples, did you mean to disable data sampling?");
    checkArgument(sampleEveryN > 0, "Expected positive number for sampling period, did you mean to disable data sampling?");
    this.maxSamples = maxSamples;
    this.sampleEveryN = sampleEveryN;
```